src/Games/NexusMods.Games.StardewValley/StardewValleyLoadoutSynchronizer.cs

```diff
@@ -52,6 +54,22 @@ protected override async Task<Loadout.Model> AddChangedFilesToLoadout(Loadout.Mo
 
             newFile.Add(File.Mod, smapiMod.Id);
             newFile.AddTo(tx);
+            modifiedMods.TryAdd<ModId, Mod.Model>(smapiMod.ModId, smapiMod);
+        }
+
+        foreach (var mod in modifiedMods.Values)
+        {
+            // If we created the mod in this transaction, db will be null, and we can't call .Revise on it
+            // We need to manually revise the loadout in that case
+            // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+            if (mod.Db != null)
+            {
+                mod.Revise(tx);
+            }
+            else
+            {
+                loadout.Revise(tx);
+            }
```

Comment:
What do you mean with

> If we created the mod in this transaction

What "mod" are you referring to? This synchronizer doesn't create any mods.

Reply:
If this is the first time you have outside changes that end up in Override, then Override is created by `GetOrCreateOverridesMod`

Reply:
If that's the case, then you should change this code to reflect that. This code suggests that the comment applies to every mod we updated.

Reply:
I updated the comment, did you want a more fundamental change to how the code works here or is that fine?

I could try to track Override specifically, but this way if we decide in the future to instead add Config files to new mods and attach those mods to existing mods or something, then this still works or at least it will be clear that any created mod needs this, not just Override.

Reply:
We don't need special handling for override mod, we need special handling for any created mod. We just happen to create override. If you want to create a another mod to put files in, then you would need to call revise on that as well.